File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        sol = 0
        for num in num_set:
            if num - 1 not in num_set:
                count = 1
                while num + count in num_set:
                    count += 1
                if count > sol:
                    sol = count
        return sol

File: Data Structures & Algorithms/split-array-largest-sum/submission-0.java
class Solution {
    public int splitArray(int[] nums, int k) {
        int l = 0;
        int r = 0;
        int res = 0;

        for(int num : nums){
            l = Math.max(l, num);
            r += num;
        }
        res = r;

        while(l <= r) {
            int m = l + (r - l) / 2;
            if(split(nums, k, m)){
                res = m;
                r = m - 1;
            }
            else{
                l = m + 1;
            }
        }
        return res;
    }

    public boolean split(int[] nums, int k, int largest){
        int subarray = 1;
        int currSum = 0;
        for(int num : nums){
            currSum += num;
            if(currSum > largest) {
                subarray++;
                if(subarray > k) return false;
                currSum = num;
            }
        }
        return true;
    }
}